### PR TITLE
Fixed bug related to displaying user posts

### PR DESCRIPTION
**Pull request by notnine**
Fixed a bug where you can view the feed from viewing a user's posts,
Changed 2 public methods in feedcontroller back to private since I am no longer using them in usercontroller

### CODE REVIEW

File: src/main/java/controllers/UserProfileController.java [hunk 1]
```diff
@@ -65,8 +62,9 @@ public void runBrowseProfile() {
         // display all the usernames in the database
         this.runDisplayAllUsers();
         String userToBrowse = this.inOut.getInput("Enter the username of the person you'd like to view: ");
-        if (this.userManager.checkUserToBrowse(userToBrowse) &&
-                !(userToBrowse.equals(this.loginManager.getCurrUser().getUsername()))) {
+        // if ((this.userManager.checkUserToBrowse(userToBrowse) &&
```
Comment (ericjest) [inline src/main/java/controllers/UserProfileController.java:65]: If we aren't using this code we should delete it, not comment it out.

File: src/main/java/controllers/UserProfileController.java [hunk 2]
```diff
@@ -226,4 +226,53 @@ private void runCustomizeBio() {
...
     }
+
+    /**
+     * Generate a selection of displayed posts' titles for the user to select.
+     *
+     * @param userPosts A hashmap of Posts representing a user's posts.
+     * @return a String representing all the posts of a user.
+     */
+    public String generateDisplayedPosts(HashMap<Integer, Post> userPosts) {
+        StringBuilder postsString = new StringBuilder("Enter a post number to view the post: ");
```
Comment (ericjest) [inline src/main/java/controllers/UserProfileController.java:237]: Are we showing the user a list of posts to select from? We can't expect the user to memorize all the post numbers.

File: src/main/java/controllers/UserProfileController.java [hunk 3]
```diff
@@ -226,4 +226,53 @@ private void runCustomizeBio() {
...
+    }
+
+    /**
+     * Get the user to select a post to act on.
+     *
+     * @param userPostsString     the String representing the selection of displayed Posts' titles generated.
+     * @return an int representing the Post that the user selects.
+     */
+    private int getSelectedPostInput(String userPostsString) {
+        int postNumber = -700;
```
Comment (ericjest) [inline src/main/java/controllers/UserProfileController.java:267]: Why -700?

**Review by ericjest — CHANGES_REQUESTED** [review]

File: src/main/java/controllers/UserProfileController.java (hunk 1 above)
Comment (notnine) [reply src/main/java/controllers/UserProfileController.java:65]: sounds good

File: src/main/java/controllers/UserProfileController.java (hunk 2 above)
Comment (notnine) [reply src/main/java/controllers/UserProfileController.java:237]: the post titles is given with the post numbers in the prompt

File: src/main/java/controllers/UserProfileController.java (hunk 3 above)
Comment (notnine) [reply src/main/java/controllers/UserProfileController.java:267]: it was just a negative number to initialize postNumber

**Pull request merged**